vault/core.go

```diff
@@ -3042,6 +3042,19 @@ func (c *Core) readFeatureFlags(ctx context.Context) (*FeatureFlags, error) {
 	return &flags, nil
 }
 
+// isMountable tells us whether or not we can continue mounting a plugin-based
+// mount entry after failing to instantiate a backend.
+//
+// We can proceed mounting without initializing the backend if the plugin
+// catalog returns no error and an external or nil plugin. We cannot proceed
+// mounting if the plugin catalog returns an error or a builtin plugin.
```

Comment:
nit: This is a really good comment, but I think it can be improved by explaining at a higher level of abstraction than the logic. The code is the source of truth for what the logic is doing anyway.

```suggestion
// isMountable tells us whether or not we can continue mounting a plugin-based
// mount entry after failing to instantiate the plugin. We do this to preserve the
// storage and path when a plugin is missing or has otherwise been misconfigured.
// It shouldn't be possible for existing builtins to be misconfigured, so that is a
// fatal error.
```